Add arguments to position summary columns and rows

JoshuaSturm opened the pull request:
- You can set the position of summary columns (left/right) and rows (above/below) using the `summaryCol` and `summaryRow` 
arguments, respectively, in `pageSetup()`.

Closes #107 

It's working in Ubuntu + LibreOffice, but corrupting the workbook in Windows + Excel. PR is in draft mode while I debug the Excel issue.

JoshuaSturm committed:
Add option to choose outline summary position

- Under `pageSetup`, you can set the position for the summary column
(summaryCol - "Left" or "Right") and summary row (summaryRow - "Above" or "Below").

JoshuaSturm committed:
Add option to choose outline summary position

- Under `pageSetup`, you can set the position for the summary column
(summaryCol - "Left" or "Right") and summary row (summaryRow - "Above" or "Below").

JoshuaSturm committed:
Merge branch 'master' into fix-issue-107

JoshuaSturm committed:
Add test and news item; updated docs.

JoshuaSturm committed:
Add test and news item; updated docs.

FrancoisR95 commented:
Great! Many thanks @JoshuaSturm for this addition.

shrektan committed:
read comments as UTF-8 encoded

shrektan committed:
add a helper function markUTF8()

shrektan committed:
add a clone argument

shrektan committed:
looks like a bug to me, should be &&

shrektan committed:
mark UTF8 when cpp functions return CharacterVector and nonASCII characters are possible involved

shrektan committed:
define a helper function readUTF8()

shrektan committed:
replace all the readLines() with readUTF8()

shrektan committed:
roxygen modifies

shrektan committed:
allows non-ASCII named regione being created

shrektan committed:
move readUTF8() to helperFunctions.R

shrektan committed:
allow non-ASCII name in writeData()

shrektan committed:
writeDataTable() support non-ASCII table names

shrektan committed:
add non-ASCII tests

shrektan committed:
add NEWS entry

ycphs committed:
Merge pull request #118 from shrektan/shrektan/encoding

UTF-8 in/out

shrektan committed:
replace 1:length() with seq_along()

ycphs committed:
Merge pull request #120 from shrektan/seq_along

replace 1:length() with seq_along()

ycphs committed:
Update DESCRIPTION

Dev version

JanMarvin committed:
Fix Warning message:
In if (grepl(":", region, fixed = TRUE)) { :
  the condition has length > 1 and only the first element will be used

JanMarvin committed:
Adjust test

JanMarvin committed:
implement reading namedRegion from specific sheet

JanMarvin committed:
be more strict, but keep backward compatible

JanMarvin committed:
`ind` might contain FALSE. a case where multiple namedRegions appear on a single sheet, but the user requested only a single one of them

JanMarvin committed:
cleanup
* the check if the Region is in the workbook has already been done
* check might complain about missing ()

JanMarvin committed:
stringsAsFactors, my old enemy!

ycphs committed:
Merge pull request #121 from JanMarvin/named_region_fix_warning

Implement reading namedRegion from specific sheet

JanMarvin committed:
bug fix: fillMergedCells and rows now work as expected

ycphs committed:
Merge pull request #122 from JanMarvin/fix_fillMergedCells

bug fix: fillMergedCells and rows now work as expected

JMPivette committed:
Add unit test on Workbook with multiple pivot tables (Issue #124)

JMPivette committed:
minor reformat of code sections titles

JMPivette committed:
fix rIds to correspond to number of Pivot Cache and not Pivot Tables

JMPivette committed:
Add unit tests loadWorkbook

ycphs committed:
Merge pull request #125 from JMPivette/fix-issue-124

Fix issue #124

shrektan committed:
setColWidths() supports 0-len cols

shrektan committed:
update NEWS

ycphs committed:
Merge pull request #129 from shrektan/setcolwidth-zero-len-cols

setColWidths() supports 0-len cols

JanMarvin committed:
do not provide a default sheet

JanMarvin committed:
The selection logic was switched a bit. Improvements are along the following lines:

* sheet names with whitespaces are now identified (might be further improved)
* reduction of index vector and sheet names vector are corrected (might add checks, or something like a struct)
* do not return something if namedRegion is found in sheet, but not on the selected sheet

JanMarvin committed:
unittests
* improve test (use sheet name as well as sheet index)
* Warning if wrong sheet was selected and no data
* Warning if wrong namedRegion was selected

JanMarvin committed:
add trailing newline when returning No data found on worksheet

JanMarvin committed:
update getChildlessNode. run twice.
1. run once to check if tag is in xml and if we are looking for <foo/> or <foo></foo>
2. do the actual run, reading everything per node

JanMarvin committed:
adapt to new format

JanMarvin committed:
initial implementation of array formulas

JanMarvin committed:
roxygen was pending

JanMarvin committed:
aint nobody got time to run check

ycphs committed:
Merge pull request #130 from JanMarvin/gh_issue_127_2

[WIP] improve getChildlessNode. fixes #127

ycphs committed:
Merge pull request #123 from JanMarvin/namedRegion

Further improvements regarding named regions

ycphs commented:
I would like to add this feature to the new release to fix the issue #132.

Do you think that's possible?

JoshuaSturm commented:
@ycphs I hope to finish this weekend.

JanMarvin committed:
clean up

JanMarvin committed:
clean up

JanMarvin committed:
completely unrelated but anoying

JanMarvin committed:
clean up

JanMarvin committed:
clean up

JanMarvin committed:
fix

JanMarvin committed:
add everything possible

JanMarvin committed:
update more headers

ycphs committed:
Merge pull request #131 from JanMarvin/gh_issue_50

[WIP] Writing Array Functions

ycphs committed:
Merge pull request #133 from JanMarvin/gh_issue_74

Update openxmlformats. Closes #74

JMPivette committed:
Add unit test for bug #126

JMPivette committed:
fix a bug that truncate comments if "/" is present

JMPivette committed:
Fix problem when saving comments with no style

JMPivette committed:
add threaded comment information

JMPivette committed:
add persons information to workbook (needed for threaded comments)

JMPivette committed:
minor changes following JanMarvin review

ycphs committed:
Merge pull request #135 from JMPivette/fix-issue-126

Fix issue #126

JoshuaSturm committed:
Fix typo in string casing

Fixed a typo in the string conditional which was causing the check to
always evaluate to false.

JoshuaSturm committed:
Merge branch 'fix-issue-107' of https://github.com/JoshuaSturm/openxlsx into fix-issue-107
